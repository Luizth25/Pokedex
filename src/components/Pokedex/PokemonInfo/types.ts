export type TPokemonInfo = {
  name: string;
  baseExperience: string;
  weight: string;
  abilities: string;
  id: string;
  children: JSX.Element;
};
