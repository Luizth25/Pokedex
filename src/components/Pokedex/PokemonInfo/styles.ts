import styled, { css } from "styled-components";

export const PokemonInfoContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: 5px;
    border-radius: 20px;
    border: 2px solid ${theme.colors.borderColor};
    background-color: ${theme.colors.textBackground};
  `}
`;

export const PokemonName = styled.h2`
  font-size: 24px;
  text-align: center;
  padding: 5px;
  font-weight: 600;
  color: ${({ theme }) => theme.colors.titleColor};
`;

export const PokemonInfo = styled.p`
  padding: 10px;
  font-size: 16px;
  color: ${({ theme }) => theme.colors.infoColor};
`;

export const InfoContainer = styled.div`
  display: flex;
  justify-content: space-around;
`;
