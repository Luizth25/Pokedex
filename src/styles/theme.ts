declare module "styled-components" {
  export interface DefaultTheme {
    colors: typeof themeColors;
  }
}

const themeColors = {
  background: "#CC322A",
  textBackground: "#E6D596",
  titleColor: "#4C3434",
  infoColor: "#642C34",
  borderColor: "#A7B9D3",
};

const theme = {
  colors: themeColors,
};

export default theme;
