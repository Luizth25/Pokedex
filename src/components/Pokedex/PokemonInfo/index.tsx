import { TPokemonInfo } from "./types";
import * as s from "./styles";

const PokemonInfo = ({
  baseExperience,
  name,
  weight,
  abilities,
  children,
  id,
}: TPokemonInfo) => (
  <s.PokemonInfoContainer>
    <s.PokemonName>
      {name} {id}
    </s.PokemonName>
    <s.InfoContainer>
      <s.PokemonInfo>{baseExperience}</s.PokemonInfo>
      <s.PokemonInfo>{weight}</s.PokemonInfo>
    </s.InfoContainer>
    <s.PokemonName>{abilities}</s.PokemonName>
    {children}
  </s.PokemonInfoContainer>
);

export default PokemonInfo;
