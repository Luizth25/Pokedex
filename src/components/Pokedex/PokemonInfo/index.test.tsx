import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import theme from "styles/theme";

import PokemonInfo from ".";

describe("The PokemonInfo Component", () => {
  it("Should be denied", () => {
    expect.assertions(1);

    expect(PokemonInfo).toBeDefined();
  });

  it("Should render component", () => {
    render(
      <ThemeProvider theme={theme}>
        <PokemonInfo
          abilities="teste 1"
          baseExperience="2"
          id="1"
          name="teste"
          weight="2"
        >
          <p>teste</p>
        </PokemonInfo>
      </ThemeProvider>
    );

    expect(screen.getAllByRole("heading", { name: /teste /i })).toHaveLength(2);
    expect(screen.getAllByText("2")).toHaveLength(2);
  });

  it("Should match snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <PokemonInfo
          abilities="teste 1"
          baseExperience="2"
          id="1"
          name="teste"
          weight="2"
        >
          <p>teste</p>
        </PokemonInfo>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
