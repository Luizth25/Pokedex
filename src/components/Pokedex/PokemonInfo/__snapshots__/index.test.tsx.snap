// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PokemonInfo Component Should match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe kGeEDM"
  >
    <h2
      class="sc-gswNZR iXYRSr"
    >
      teste
       
      1
    </h2>
    <div
      class="sc-hLBbgP Zpgwu"
    >
      <p
        class="sc-dkrFOg jdRPmV"
      >
        2
      </p>
      <p
        class="sc-dkrFOg jdRPmV"
      >
        2
      </p>
    </div>
    <h2
      class="sc-gswNZR iXYRSr"
    >
      teste 1
    </h2>
    <p>
      teste
    </p>
  </div>
</div>
`;
